### ISSUE TITLE
Additional minor fixes

### ISSUE THREAD

**Issue by mchaindl** [EPFL-ENAC/eesd-mast#15]
@ymarcon @charlottegiseleweil These are some minor fixes required to be applied (low priority fixes)

### **Overview tab**
- [x] The kernel function (curve displayed over the points in the PGA vs. DG plot), should be truncated at the lower and upper points.

- [x] Remove (DG) in the y-label, and keep it as "Damage grade" only.

- [x] In x-label, put the units in brackets. Instead of (g) please use [g units].

- [x] Keep always the complete parallel plot and highlight the connecting paths according to the filtering in the donnut charts.

### **Detail window of each building**
- [x] In case the evolution of the fundamental period is not available, put "Evolution of fundamental period not available" in the space where the plot should appear.

- [x] Change "Evolution of fund. period" by "Measured fund. period".

- [x] Change the y-label of the fundamental period plot by "Measured of fund. period in X-dir. [sec.]" or "Measured of fund. period in Y-dir. [sec.]", depending on for which direction the information is available. Same for x-label, it should be "Measured (or Nominal) PGA at the shake table in X-dir [g units]".

- [x] Delete (DG) in x-label of plots. Leave it as "Damage grade" only.

- [x] Show only one plot for Damage grade vs. PGA. In case PGAs are given in both X and Y directions, include them all in the same plot and then that information is given once you place the pointer on top of the dots.

- [x] In the plot Measured fund. period vs. PGA at the shake table, see if it is possible to plot the fundamental periods only if the next PGA is greater than the previous one. For example, if the pairs of data are: [PGA = 0.1g, T1 = 0.2sec.], [PGA = 0.2g, T1 = 0.23sec.], and [PGA = 0.18g, T1 = 0.22sec.] (in this order), then only the first and second pairs are plotted because the third one corresponds to a PGA that is lower than the previous one. 

- [x] Regarding the previous comment, include a note right below the plot that says: "Note: fundamental periods are plotted here only for increasing values of PGA along the test".

- [x] In the wall thickness info, sometimes there is more than one value that for the moment are shown as separated by a "/" symbol. Please replace it by "and" (for example, for the building Senaldi et al. (2020) it is shown as 250/300/350 and, if possible, it should be 250, 300, and 350.

### **Contact us tab**
- [x] Above the contact, please put "If you want to give us feedback on the content and usability of this platform, or want to share new experimental data to be included, please contact us:"

(moved from https://github.com/EPFL-ENAC/eesd-mast/issues/15#issuecomment-2112038399)

### **Overview tab**
- [x] Add a title for the donut charts. It could be "Features of the buildings included in the database". This could be right above the info note "Select a category to filter tested buildings"

- [x] The welcome pop-up message appears only once, please show it every time you go back to the Overview tab. Or let's discuss if it is possible to show it always somewhere.

- [x] Change the welcome message to: Welcome to this web-based platform dedicated to shake table tests conducted on unreinforced masonry buildings!
Here you will be able to explore our comprehensive repository and discover valuable data and insights derived from experimental tests.
Whether you are delving into seismic performance assessment or structural dynamics, our database offers a unique perspective on the seismic behavior of unreinforced masonry structures under diverse conditions.
Join us on our mission to bolster earthquake resilience and deepen our understanding of these important structures, paving the way for safer communities and more resilient infrastructure. Welcome aboard!

- [x] Truncate the fragility curves at 1g (show only up to 1g).

- [ ] ~~Add a note for the plot of fragility curves saying "Damage grades evaluated according to the EMS-98 methodology"~~ (note: added reference instead)

- [x] Highlight the "Vulnerability" section and maybe rename it to "Check vulnerability analysis".

### **Buildings tab**
- [x] Add a legend for the buildings that have the numerical model. Right below the slider ("Show numerical models") we can put the icon and the legend that can say "Buildings with numerical model available".

- [ ] ~~Add "Numerical model available" as another filter to show only those buildings with numerical models.~~ (note: ok to have only toggle on top of table)

- [x] Instead of just putting "Buildings" at the beginning of the Buildings tab, please put the following text: Here, you will find a comprehensive list of shake table tests conducted on unreinforced masonry buildings. You can filter these tests (using the filter section located left-side) based on various characteristics of the buildings, such as construction materials, building geometry, and testing considerations. Once you select a building from the list, you will be able to view detailed experimental results, including general information from the shake table tests.

- [x] I found a bug when filtering. For example, when filtering based on "Test Scale", if you start filtering with "1" only and then you select "1/2", no additional buildings are added to the list.

### **Building's pop-up window (when clicking on one of the buildings)**
- [x] Check the space of the reference given for each experiment. It seems it is now justified so the space between words increases.

- [x] Please conserve the Reference tab shown before the last update of the database (not the "Details" tab because it will be available again when going to the experimental results).

- [x] Put the link buttons "View test" and "View numerical model" right above the images of the buildings.

- [ ] ~~Make the image of the test and the numerical model the same size (at least the same height if possible)~~ (note: same width is more web friendly)

### **Experimental results window for each building (when clicking on the "View details", which, according to the last meeting, will now be "See experimental results")**
- [x] We still need to modify the x-label of the PGA vs. Period plot. It should be PGA measured at the shake table in the X-dir. [g units] (or Y-dir. depending on the building). In case only the nominal PGA is given, this axis should be "Nominal PGA in X-dir. (or Y-dir.) [g units]".

- [x] Align the PGA vs. Period plot with the Damage grade vs. PGA plot. For example, in the Beyer et al. (2015) building these plots are not aligned vertically. Make them the same size vertically.

- [x] The link to the paper for each experiment should be green (in the pop-up window of each building and also when going to the detailed test results of each building).

- [x] Consider always 3 significant digits for all the numbers in the table given in "TEST RESULTS", except for the column related to DG for which it should be shown as an integer.

- [x] Change the headers of the columns in "TEST RESULTS":
Run ID -> Run ID (stays the same)
Nominal PGA X -> Nominal PGA X-dir. [g] (similar when PGA exists for Y or Z directions -> Y-dir. and Z-dir, respectively)
Actual PGA X -> Actual PGA X-dir. [g] (similar when PGA exists for Y or Z directions -> Y-dir. and Z-dir, respectively)
DG Reported -> DG Reported (stays the same)
Max Top Drift X -> Max Top Drift X-dir. [%] (similar when exists for Y direction -> Y-dir.)
Residual Top Drift X -> Residual Top Drift X-dir. [%] (similar when exists for Y direction -> Y-dir.)
Base Shear Coefficient -> Base Shear Coefficient (stays the same)
Reported T1 X -> Measured fund. period X-dir [sec.] (similar when exists for Y direction -> Y-dir.)

- [x] Rename the "FILES" section for "TEST FILES".

### **Details of buildings (numerical models)**
- [x] Make the vtk visualizer window a little bit smaller (more close to the size of the image of the model of the building on the left).

- [x] Please change and add the following in the modeling assumption of all numerical models:
![Fixes_NumericalModels](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/9587206b-083d-4ad2-972c-09be055b7680)

- [x] Replace the static picture of the model with .vtk model and show it directly at the top of the window, when entering the numerical model. This means that the "3D MODEL" section is no longer needed. I will not give the option of downloading the .vtk file.

- [x] #28
https://kitware.github.io/glance/app/
https://github.com/Kitware/paraview-visualizer
https://www.youtube.com/watch?v=UcFWOzG5ygM
This way it would be possible to conserve the different colors for the different elements in the model.

- [x] Rename the 'FILES" section for "OPENSEES MODEL".

- [x] Rename the "DETAILS" section for "MODELING ASSUMPTIONS". Here it should appear the corresponding tables that I uploaded to the shared folder.

(moved from https://github.com/EPFL-ENAC/eesd-mast/issues/15#issuecomment-2116857795)

### Overview tab
=> latest request moved into #26 
- [x] #26

- [x] #27
![OverviewTab_FixAlignment](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/afa35d55-e1d7-4535-ac08-c678a14fd3d9)

![OverviewTab_FixAlignment2](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/37d95d89-64de-4a5a-a1eb-045b7a1e7806)

- [ ] To be consistent with the "Buildings" page, let's use the ? icon to show what can be done on the "Overview" page. This way we can save a little bit of space and maybe pull a little bit up the plots:
![OverviewTab_PopUp_Info](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/86ce6d3b-cc5a-4a2b-a67d-fe32eb9b562e)
The text to display could be: "You can select a feature of the buildings to see how they are related to other features. This selection also serves as a filter when you go to the Buildings section (where you can find the complete list of buildings included in this database)
Furthermore, based on your feature selection, you can check the results of a vulnerability analysis by using the toggle "Show vulnerability plots""

- [ ] Add "Contact Us" as a title for the Contact pop-up window:
![Add_ContactUs](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/b8c7cf8a-0de2-4e35-aeb5-785814c5e8d8)

- [x] We need to add a header image and title in the overview tab. The title could be "Database of shake table tests and models of URM buildings" or "Open-access shake table test repository for URM buildings". I will design a background image to be included with this title.
- [x] In the shared folder (ENAC-IT --> Data --> Archive) I put an example of the background we can use for the top of the initial page of the platform. I will put the title on top of it. 
Here is one example:
![WebsiteTitle](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/10684611-a5c3-432a-81b5-5bdad420500f)

But maybe we can try something like this (I think it looks nice, what do you think?. The "Contact" is replaced by an icon as well);
![Website_OverviewTab_NewDesign_v2](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/dd70f0a9-fcb2-4202-a1a8-69c54e55602a)

- [x] ~~Is it possible to show at least once (for example, when going for the first time to the website) the welcome pop-up message?~~ (note: that is already the case, the "first" visit is tracked in a cookie, you can either remove the cookie or open the page in an anonymous window to see what it looks like)

- [x] Please increase the fontsize of "Overview" and "Buildings" tab links. Same for "Features of the buildings included in the database". 

- [x] The button to show the vulnerability plots should in in green color as it is related to the experimental results.

- [x] Make the fontzise of the labels in the parallel plot a little bit bigger and, if possible, push the plot a little bit upper.

- [x] If possible, make the color of the selected feature in the parallel plot the same as in the donut charts. For example, if I choose "Stone", the linked features in the parallel plot are shown in blue color.

- [ ] ~~When showing the fragility curves, if possible, put the legend as follows:~~ (note: adding a legend group title is not supported by plotly)**(note: In that case, please only show the legend for the fitted curves)**
![Fixes_FragilityCurves](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/22ec60b1-21a8-4d0c-959f-638a59b36cbd)

- [x] The fragility curves should appear even if a DG is not available. For example, if fragility curves can be computed only for DG1, DG2, and DG3, these three curves should be displayed (even if no curves are generated for DG4 and DG5).

- [x] The visibility of the fragility curves in the plot must be truncated to 1g, but not the curves themselves. By default, I should see up to 1g, but if I scroll the curves to the right, I should still be able to see the curves for PGA greater than 1g.

- [x] The vulnerability curves are not considering properly all the buildings according to the filters. For example, if I select the stone buildings I should see three points for DG5 (1.051g, 0.34g, and 0.535g), but only one is shown. Something is missing there.

- [x] In the vulnerability section, in the PGA vs. Damage grade plot (left plot) we need to show always the actual PGA unless only the nominal PGA is available.

- [x] It would be nice to show the empirical fragility curves (a do for each point) and also the fitted fragility curve (as shown now).

- [x] We need to include a note below the fragility curves plot with the reference to the paper that describes the methodology to compute the curves. We can put: "Fragility curves derived according to Baker, J. W. (2015). Efficient analytical fragility function fitting using dynamic structural analysis. 465 Earthquake Spectra, 31(1), 579–599. https://doi.org/10.1193/021113EQS025M"

### Buildings tab
- [ ] <s>If possible, freeze the upper part so it is always available even when scrolling down: </s> **(moved to new issue)**
![BuildingsTab_FreezeTop](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/a2420e05-f2be-44a9-95b9-5a29ab4a94d2)

- [ ] <s>Add extra info to the pop-up window that appears when clicking the ? icon:</s> **(moved to new issue)**
![BuildingsTab_PopUp_Info](https://github.com/EPFL-ENAC/eesd-mast/assets/80859197/385ac8f9-4efe-4eef-a78b-4661f007b97b)
The text to be added is: "In addition, by using the toggle “Show only buildings with numerical models” you can check the buildings for which a numerical model is available."

- [x] According to the color scheme, the filters should go in red (if I understand that correctly)

- [x] I noticed that, when applying and changing some filters, not always the data is refreshed correctly. For example, if I select first "Clay" as "Masonry Unit Material" and then I select "Stone" (with the "Clay" also selected), and then I remove the "Clay", I still see the "Clay" buildings in the list.

- [x] When going into the numerical models, the Costa et al. (2013) model should not exist, I haven't created it.

- [x] There are buildings (for example the buildings of Candeias et al. 2017) for which the plan view is shown twice. There should be only one plan view per building.

- [x] In the "Details" section, delete the "Material Characterization References" and leave only the "Links to Material Characterization Papers".

- [x] In the "Details" section, delete the "Open Measured Data" and leave only the "Link to Open Measured Data". If the data is not openly available, put in this field "Data not openly available. Please contact the authors of the test".

- [x] In the filter for the "Masonry Unit Material", we need to merge all stone-type masonry (Calcareous sandstone, Granite, Limestone, Tuff stone, Calcareous tuff stone, Neopolitan tuff stone) as "Stone" (same as we did in the Overview tab for the donut chart).

**Comment by mchaindl** [comment]
@ymarcon @charlottegiseleweil Here are some additional comments from my last revision:

(note: moved to issue head)

**Comment by ymarcon** [comment]
Where can I find these modeling assumptions? Are these the fields grouped under "Digitalized data available" name, with Yes/No values in the Excel file?

**Comment by mchaindl** [comment]
Since this is part of the numerical models, it is not in the same Excel file as the experimental data.
I created a new Excel file with these modeling assumptions. The file is called "Modeling assumptions.xlsx" and is stored in the folder 10_Mock-up for window of numerical model.
Let me know if you cannot find it.

**Comment by ymarcon** [comment]
Ah ok, I missed that one. I just moved it to the folder 00_MAST_Database where all the data files are.

**Comment by mchaindl** [comment]
@ymarcon I have some additional comments regarding the last version of the platform:

(moved to issue head)

**Comment by ymarcon** [comment]
@mchaindl do you also want to merge the mixed diaphragm materials (RC & Timber, Steel & bricks, Hollow tile slab) in the Buildings filter, like in the Overview?

**Comment by ymarcon** [comment]
@mchaindl regarding the buildings with several plans, it is because you provided several

![image](https://github.com/EPFL-ENAC/eesd-mast/assets/2581149/e98c89dd-1e63-450c-823f-2f2d9475bb15)

Should I remove the Brick/Stone variants?

**Comment by mchaindl** [comment]
@ymarcon 
Regarding the first question (merge mixed diaphragms), it is not necessary for that case. I will leave it as it is now.

For the second question (plan views), please remove the ones that do not contain Brick/Stone in the name. Use the ones named Brick and Stone for the corresponding buildings.

**Comment by ymarcon** [comment]
@mchaindl Regarding your comment about the Vulnerability left plot: "if I select the stone buildings I should see three points for DG5 (1.051g, 0.34g, and 0.535g)". Are you sure for the 0.34g one? I have fixed the pga calculation and the values are considered in this order (ie if missing, take the next one):

1. actual pga x
2. nominal pga x
3. actual pga y
4. nominal pga y

and now there are only PGAs 1.051g and 0.535g for Stone buildings at DG5

**Comment by mchaindl** [comment]
@ymarcon According to what I see from the database, there should be one point for the test Candeias et al. (2017) (1.051g), one for Dolce et al. (2008) (0.34g), and one for Costa et al. (2013) (0.535g). Is that the same in the version of the database you have?

**Comment by ymarcon** [comment]
@mchaindl thanks, now it is fixed

**Comment by charlottegiseleweil** [comment]
moved the rest to #26
